### ISSUE TITLE
after call setupDataSource,the plugin will hang on in loading status ,until click button to play[BUG]

### ISSUE THREAD

**Issue by saviourdog** [jhomlala/betterplayer#423]
after call setupDataSource,the plugin will hang on in loading status ,until click button to play

both android and ios

here is mini code to show this problem,best regrades!
https://github.com/saviourdog/bpbug

**Comment by jhomlala** [comment]
@saviourdog Please use issue template next time. I've used your code and added one line: 
```dart
bpCon(String url, {cache}) => BetterPlayerController(
    ...
      autoPlay: true,
   ...
```
With that line, video plays after selecting it from files. If this won't help you, can you record video where you'll present that bug?

**Comment by jhomlala** [comment]
@saviourdog Can you confirm if it solved your issue?

**Comment by jhomlala** [comment]
No activity from author. Closing.

**Comment by saviourdog** [comment]
hi @jhomlala  ,Sorry to bother you again,but i think it dont slove my issue,after call setDataSource without autoplay,the expected behaviour is BetterPlayer widget show first frame of video(same with BetterPlayer first initialize),but current behaviour is   BetterPlayer widget hang on in loading status,until the video play.
here is video https://youtube.com/shorts/KguEbY--JvE?feature=share

**Comment by schisms** [comment]
I had the same issue. I did not open an issue because I use a custom theme and thought I was the problem. In my instance, I use autoplay, and the loading widget would continue to appear **after** the video was playing. 

Having a timer on my custom theme, I simply used: 
`if (_isLoading && controller.isPlaying()) => setState()`

bool _isLoading value is set on build method, so on set state it would set isLoading to false as video is done loading.
